sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -88,16 +88,27 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
         s"Selected $selected partitions out of $total, pruned $percentPruned% partitions."
       }
 
+      // need to add projections from combineFilters in
+      val combineFilter = combineFilters.reduceLeftOption(expressions.And)
+      val combinedProjects = combineFilter.map(_.references.toSet.union(projects.toSet).toSeq)
```

Comment:
Don't call `toSet`.  Anything involving attributes set logic should be done with an `AttributeSet` (which ignores cosmetic differences like capitalization).